api/v1alpha1/databasecluster_types.go

```diff
@@ -116,6 +150,20 @@ type Engine struct {
 	UserSecretsName string `json:"userSecretsName,omitempty"`
 }
 
+// Size returns the size of the engine.
+func (e Engine) Size() EngineSize {
+	if e.Resources.Memory.Cmp(MemorySmallSize) == 0 {
+		return EngineSizeSmall
+	}
+	if e.Resources.Memory.Cmp(MemoryMediumSize) == 0 {
+		return EngineSizeMedium
+	}
+	if e.Resources.Memory.Cmp(MemoryLargeSize) == 0 {
+		return EngineSizeLarge
+	}
+	return EngineSizeUnknown
```

Comment:
I agree the intervals make more sense than the discrete values, since there are the valid values in between. However I may miss something.  

if 2Gb is a minimum requirement, then:

2Gb <= mem < 4Gb        - small
4Gb <= mem < 32 Gb     - medium
32Gb <= mem                - large

Reply:
Yeah, I think intervals make more sense, since it will also cover the case with the discreet values.
I have updated the code now